.github/workflows/ci.yml

```diff
@@ -34,119 +34,86 @@ jobs:
-        run: "./pathfinder_test --skip ethereum::"
-
-  test_alchemy:
-    name: pathfinder tests (alchemy)
-    needs: compile_tests
+  test_pathfinder_ethereum_alchemy:
+    name: pathfinder-ethereum tests (alchemy)
+    needs: compile_pathfinder_ethereum_tests
     runs-on: ubuntu-latest
     steps:       
       # Fetch the test binary artifact
       - uses: actions/download-artifact@v3
         with:
-          name: test-binary
+          name: test-binary-pathfinder-ethereum
       # Archiving removes the file permissions
       - name: Fix file permissions
-        run: chmod +x pathfinder_test
+        run: chmod +x pathfinder_ethereum
 
       - name: Execute tests
         run: |
           PATHFINDER_ETHEREUM_HTTP_GOERLI_URL=${{ secrets.ALCHEMY_GOERLI_URL }} \
             PATHFINDER_ETHEREUM_HTTP_GOERLI_PASSWORD=${{ secrets.ALCHEMY_GOERLI_PASSWORD }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_URL=${{ secrets.ALCHEMY_MAINNET_URL }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_PASSWORD=${{ secrets.ALCHEMY_MAINNET_PASSWORD }} \
-            timeout 3m ./pathfinder_test ethereum::
+            timeout 3m ./pathfinder_ethereum
 
-  test_infura:
-    name: pathfinder tests (infura)
-    needs: compile_tests
+  test_pathfinder_ethereum_infura:
+    name: pathfinder-ethereum tests (infura)
+    needs: compile_pathfinder_ethereum_tests
     runs-on: ubuntu-latest
     steps:       
       # Fetch the test binary artifact
       - uses: actions/download-artifact@v3
         with:
-          name: test-binary
+          name: test-binary-pathfinder-ethereum
       # Archiving removes the file permissions
       - name: Fix file permissions
-        run: chmod +x pathfinder_test
+        run: chmod +x pathfinder_ethereum
 
       - name: Execute tests
         run: |
           PATHFINDER_ETHEREUM_HTTP_GOERLI_URL=${{ secrets.INFURA_GOERLI_URL }} \
             PATHFINDER_ETHEREUM_HTTP_GOERLI_PASSWORD=${{ secrets.INFURA_GOERLI_PASSWORD }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_URL=${{ secrets.INFURA_MAINNET_URL }} \
             PATHFINDER_ETHEREUM_HTTP_MAINNET_PASSWORD=${{ secrets.INFURA_MAINNET_PASSWORD }} \
-            timeout 3m ./pathfinder_test ethereum::
+            timeout 3m ./pathfinder_ethereum
 
-  test_stark_curve:
-    name: stark_curve tests
+  test_all_except_pathfinder_ethereum:
+    name: all tests w/o pathfinder-ethereum
     runs-on: ubuntu-latest
     steps:
+      # Required for the python environment
       - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - name: tests
-        run: cargo test -p stark_curve --all-targets --locked
 
-  test_stark_hash:
-    name: stark_hash tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - name: tests
-        run: cargo test -p stark_hash --all-targets --locked
```

Comment:
How does this even work without the rust-toolchain step? The rust-cache thingy seems to be missing too.

Reply:
That was a good catch and yet when I look at the logs miraculously cargo is there even though the toolchain action is missing.